advanced.matrix/src/main/java/learning/solutions/advanced/matrix/domain/RCell.java

```diff
@@ -60,6 +64,40 @@ public double getMeanLong() {
         return meanLong;
     }
 
+    private List<Long> rewardTimestamps = new ArrayList<>();
+    private List<Long> penaltyTimestamps = new ArrayList<>();
+
+    public double recalculateRewardValue() {
+        Long currentTime = System.currentTimeMillis();
+        purgeExpiredPenaltyTimes(currentTime);
+        purgeExpiredRewardTimes(currentTime);
+        return reward + (rewardTimestamps.size() - penaltyTimestamps.size()) * 10.0d;
```

Comment:
This will not work if the positive reward and negative reward are different values. Please make constants for these respectively - which can be used as default values, but they should come from a properties file - so that we can set reward and punishments without having to make code changes. Also the delay or threshold - should come from properties file.